src/main/scala/circt/Convention.scala

```diff
@@ -0,0 +1,52 @@
+package chisel3.circt
+
+import chisel3.RawModule
+import chisel3.experimental.{annotate, ChiselAnnotation}
+import firrtl.annotations.{ModuleName, SingleTargetAnnotation}
+import firrtl.FirrtlUserException
+
+/** Annotation to force the compiler to use a specified port convention.
+  * May only be applied to modules.
+  */
+case class ConventionAnnotation(target: ModuleName, convention: String) extends SingleTargetAnnotation[ModuleName] {
+  def duplicate(target: ModuleName): ConventionAnnotation =
```

Comment:
Nit: `override def` is better Scala style as it indicates that you are overriding something/implementing an abstract member. Unfortunately there is no way to strictly enforce this. 🥲